libr/asm/p/asm_mips_gnu.c

```diff
@@ -43,8 +45,37 @@ static int disassemble(struct r_asm_t *a, struct r_asm_op_t *op, const ut8 *buf,
 	Offset = a->pc;
 	memcpy (bytes, buf, 4); // TODO handle thumb
 
+	if ((a->cpu != pre_cpu) && (a->features != pre_features)) {
+		free (disasm_obj.disassembler_options);
+		memset (&disasm_obj, '\0', sizeof (struct disassemble_info));
+	}	
+
 	/* prepare disassembler */
-	memset (&disasm_obj,'\0', sizeof (struct disassemble_info));
+	if (a->cpu && a->cpu != pre_cpu) {
+		if (!r_str_casecmp (a->cpu, "mips64r2")) {
+			disasm_obj.mach = bfd_mach_mipsisa64r2;
+		} else if (!r_str_casecmp (a->cpu, "mips32r2")) {
+			disasm_obj.mach = bfd_mach_mipsisa32r2;
+		} else if (!r_str_casecmp (a->cpu, "mips64")) {
+			disasm_obj.mach = bfd_mach_mipsisa64;
+		} else if (!r_str_casecmp (a->cpu, "mips32")) {
+			disasm_obj.mach = bfd_mach_mipsisa32;
+		}
+		pre_cpu = a->cpu;
+	}
+
+	if (a->features && a->features != pre_features) {
+		disasm_obj.disassembler_options = r_str_new("");
+		if (strstr (a->features, "n64")) {
+			r_str_append (disasm_obj.disassembler_options, "abi=n64,");
```

Comment:
this will generate a string with trailing comma. its better to prefix the comma in case the string is not !=.. also this is not in a loop, and check cases have an else.. so i think that there will be only one of these cases set.

Reply:
disasm_obj.disassembler_option may provide users with an interface that can set ABI ()
mips-dis.c:
print_mips_disassembler_options (FILE *stream)
I think there may be other options that will be added by the user